tests/ludwig/config_validation/test_checks.py

```diff
@@ -198,3 +198,33 @@ def test_dense_binary_encoder_0_layer():
     }
     with pytest.raises(ConfigValidationError):
         ModelConfig.from_dict(config)
+
+
+@pytest.mark.parametrize(
+    "entity_1,entity_2,expected",
+    [
+        (["a1"], ["b1"], True),
+        (["a1", "a2"], ["b1"], True),
+        (["a1", "b1"], ["b1"], False),
+    ],
+)
+def test_comparator_combiner(entity_1: List[str], entity_2: List[str], expected: bool):
+    config = {
+        "input_features": [
+            {"name": "a1", "type": "category"},
+            {"name": "b1", "type": "category"},
+        ],
+        "output_features": [
+            {"name": "out1", "type": "binary"},
+        ],
+        "combiner": {
+            "type": "comparator",
+            "entity_1": entity_1,
+            "entity_2": entity_2,
+        },
+    }
+
+    with pytest.raises(ConfigValidationError) if not expected else contextlib.nullcontext():
```

Comment:
Maximally parameterizing tests to minimize code duplication is tempting, but it can make tests more difficult to read and understand when it involves adding logic to the test itself.

See ["Avoid logic in tests"](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-logic-in-tests) from Microsoft's testing best practices guide.

The behavior we are testing for is more readable/understandable as three separate tests, imo:

```python
def test_comparator_combiner_raises_missing_features():
def test_comparator_combiner_removes_extra_features():
def test_comparator_combiner_raises_duplicated_features():
```

Reply:
Can't say I agree. I was thinking of adding a handful of additioanl tests to this. Parametrizing makes this super easy. Separate tests means in practice, no one's ever going to bother.